core/src/main/java/cucumber/runner/TestCaseSyncEventBus.java

```diff
@@ -0,0 +1,69 @@
+package cucumber.runner;
+
+import cucumber.api.event.Event;
+import cucumber.api.event.EventHandler;
+import cucumber.api.event.TestCaseEvent;
+import cucumber.api.event.TestCaseFinished;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestCaseSyncEventBus implements EventBus {
```

Comment:
Can you split this into a `TestCaseEventBus` that queues up `TestCase` events and a `SynchronizedEventBus` that ensures only one thread can send events at the same time. This may require adding the `sendAll` method to the `SynchronizedEventBus`.

Reply:
Done